### ISSUE TITLE
cAdvisor queries missing containers over and over

### ISSUE THREAD

**Issue by colinmollenhour** [kontena/kontena#2700]
Looking at /var/log/syslog I see that kontena appears to query for lots of containers on a given node that don't exist..

```
$ grep "No such container" /var/log/syslog | wc -l
5352
```

```
Aug 16 06:25:07 admin1 dockerd[27499]: time="2017-08-16T06:25:07.442105724-05:00" level=error msg="Handler for GET /containers/fa1b34574dafe559c51ff6e3e671b84e3bbb376239459baf107b689e0c6bf5ff/json returned error: No such container: fa1b34574dafe559c51ff6e3e671b84e3bbb376239459baf107b689e0c6bf5ff"
Aug 16 06:25:07 admin1 dockerd[27499]: time="2017-08-16T06:25:07.442652229-05:00" level=error msg="Handler for GET /containers/2681561988a1c3c969186a716d577c54aa878e86e5456bc838e5750cfbfe48ed/json returned error: No such container: 2681561988a1c3c969186a716d577c54aa878e86e5456bc838e5750cfbfe48ed"
Aug 16 06:25:07 admin1 dockerd[27499]: time="2017-08-16T06:25:07.443318034-05:00" level=error msg="Handler for GET /containers/59bd9ac2c0dd3a84b23b432f39a67cf7292a08901217e213d8966bee9c5f6658/json returned error: No such container: 59bd9ac2c0dd3a84b23b432f39a67cf7292a08901217e213d8966bee9c5f6658"
Aug 16 06:25:07 admin1 dockerd[27499]: time="2017-08-16T06:25:07.443906785-05:00" level=error msg="Handler for GET /containers/526397d73bad5757d3bac410928be5b29aca9445335d2e0041f8e0f9260ec984/json returned error: No such container: 526397d73bad5757d3bac410928be5b29aca9445335d2e0041f8e0f9260ec984"
Aug 16 06:25:07 admin1 dockerd[27499]: time="2017-08-16T06:25:07.444309715-05:00" level=error msg="Handler for GET /containers/e30281e00c1c6c8042b97857c1a3bbd7bac160ae154bdf06af9efb515bbbd14a/json returned error: No such container: e30281e00c1c6c8042b97857c1a3bbd7bac160ae154bdf06af9efb515bbbd14a"
Aug 16 06:25:07 admin1 dockerd[27499]: time="2017-08-16T06:25:07.444965583-05:00" level=error msg="Handler for GET /containers/5948f01fc12c7b4169ed6a164953bb4f27d3cf6e651dd792c2994512bf5c6130/json returned error: No such container: 5948f01fc12c7b4169ed6a164953bb4f27d3cf6e651dd792c2994512bf5c6130"
Aug 16 06:25:07 admin1 dockerd[27499]: time="2017-08-16T06:25:07.445661768-05:00" level=error msg="Handler for GET /containers/94fb2e4e818de7155033760c28c59ec4a2143de92c885dd73d4c3d7a766a485f/json returned error: No such container: 94fb2e4e818de7155033760c28c59ec4a2143de92c885dd73d4c3d7a766a485f"
Aug 16 06:25:07 admin1 dockerd[27499]: time="2017-08-16T06:25:07.446246579-05:00" level=error msg="Handler for GET /containers/abaee58ffe23f96703a9bab1d5c87ea961c6206abc5bb2d9d09479f4293a758f/json returned error: No such container: abaee58ffe23f96703a9bab1d5c87ea961c6206abc5bb2d9d09479f4293a758f"
Aug 16 06:25:07 admin1 dockerd[27499]: time="2017-08-16T06:25:07.446844069-05:00" level=error msg="Handler for GET /containers/c0e30249844ddfb3c249f8a998d7b9f36144b6b81ed860061c8d57bb4edb02d7/json returned error: No such container: c0e30249844ddfb3c249f8a998d7b9f36144b6b81ed860061c8d57bb4edb02d7"
Aug 16 06:25:07 admin1 dockerd[27499]: time="2017-08-16T06:25:07.447698456-05:00" level=error msg="Handler for GET /containers/30b661db129d2cbc889392ec8a5b6bc1c1d68c7805a8f4139941048acb94ec17/json returned error: No such container: 30b661db129d2cbc889392ec8a5b6bc1c1d68c7805a8f4139941048acb94ec17"
Aug 16 06:25:07 admin1 dockerd[27499]: time="2017-08-16T06:25:07.448471852-05:00" level=error msg="Handler for GET /containers/d66c6eb5cace4ac660a6db7482f6999cc5a08007c3996cae0632f06b5f846c96/json returned error: No such container: d66c6eb5cace4ac660a6db7482f6999cc5a08007c3996cae0632f06b5f846c96"
Aug 16 06:25:07 admin1 dockerd[27499]: time="2017-08-16T06:25:07.448909607-05:00" level=error msg="Handler for GET /containers/51921edef6a3ebeaa7e57be4106dbf67dee5751017a58bba366845e2bbddee5f/json returned error: No such container: 51921edef6a3ebeaa7e57be4106dbf67dee5751017a58bba366845e2bbddee5f"
```

I finally started just dropping these via syslog config but it seems like this could indicate a good opportunity for improving Kontena's resource efficiency.. My kontena server burns 50% CPU pretty much constantly even if I haven't updated a single image or service for months.

**Comment by SpComb** [comment]
I recall running into these while debugging the cadvisor issues (https://github.com/google/cadvisor/issues/1572), and my guess is that these spurious container gets are actually from cadvisor: https://github.com/google/cadvisor/issues/1443 https://github.com/kubernetes/kubernetes/issues/38979#issuecomment-268340625

> * The docker handler in cadvisor notices that there are at least 64 alphanumeric characters in a row, so technically maybe there's a docker id in there (spoiler there's not).
> * cAdvisor inspects the ID it captured to see if docker agrees it's a container
> * docker doesn't recognize it and logs an error

We should probably try the newer cadvisor releases, v0.25.0 contains a fix for the `.mount` cgroups, and that might also help with this.

**Comment by SpComb** [comment]
Should be fixed in 1.4.2 by #3115

Re-open if not.
